go/test/endtoend/keyspace/keyspace_test.go

```diff
@@ -247,13 +252,18 @@ func TestDeleteKeyspace(t *testing.T) {
 	// Start over and this time use recursive DeleteKeyspace to do everything.
 	_ = clusterForKSTest.VtctldClientProcess.CreateKeyspace("test_delete_keyspace", sidecar.DefaultName)
 	_ = clusterForKSTest.VtctldClientProcess.ExecuteCommand("CreateShard", "test_delete_keyspace/0")
-	_ = clusterForKSTest.VtctlclientProcess.ExecuteCommand("InitTablet", "--", "--port=1234", "--bind-address=127.0.0.1", "--keyspace=test_delete_keyspace", "--shard=0", "zone1-0000000100", "primary")
+	_ = clusterForKSTest.InitTablet(&cluster.Vttablet{
+		Type:      "primary",
+		TabletUID: 100,
+		Cell:      "zone1",
+		HTTPPort:  1234,
+	}, "test_delete_keyspace", "0")
 
 	// Create the serving/replication entries and check that they exist,
 	//  so we can later check they're deleted.
 	_ = clusterForKSTest.VtctldClientProcess.ExecuteCommand("RebuildKeyspaceGraph", "test_delete_keyspace")
-	_ = clusterForKSTest.VtctlclientProcess.ExecuteCommand("GetShardReplication", cell, "test_delete_keyspace/0")
-	_ = clusterForKSTest.VtctldClientProcess.ExecuteCommand("GetSrvKeyspaces", "test_delete_keyspace", cell)
+	_, _ = clusterForKSTest.TopoProcess.Server.GetShardReplication(context.Background(), cell, "test_delete_keyspace", "0")
```

Comment:
Why not use the `ShardReplicationPositions` vtctldclient command here and below? Is it not equivalent?

Reply:
🤦 simply me grepping too exactly (started with "Get")

Reply:
added in 2014 as part of ["more test cleanup"](https://github.com/vitessio/vitess/commit/8f608994821c6eb7ade6fa79574d96ab65546cfb) with no further details (committed directly to main, no PR).

Looking at the [message definition](https://github.com/vitessio/vitess/blob/main/proto/topodata.proto#L305-L315), it's not ... super instructive? From the [usage of the topo method](https://github.com/search?q=repo%3Avitessio%2Fvitess+%22.GetShardReplication%22+NOT+language%3A%22Protocol+Buffer%22+NOT+path%3A%2F%5Ego%5C%2Fvt%5C%2Fvtadmin%5C%2F%2F+NOT+path%3A%2F%5Eweb%5C%2F%2F&type=code), it's just used to get a list of tablets in a shard (without needing to start at the shard primary, and then walk the replication graph manually).

This is updated during tablet init via [`UpdateTabletReplicationData`](https://github.com/vitessio/vitess/blob/15f58869c45e47875f26e95291bcc656a686a269/go/vt/topo/tablet.go#L506) and conversely with [`DeleteTabletReplicationData`](https://github.com/vitessio/vitess/blob/15f58869c45e47875f26e95291bcc656a686a269/go/vt/topo/tablet.go#L511) to remove a node.

Given that we include `ShardReplication{Add,Fix,Remove}` to manually edit the topo data, it would be useful to "get" that data to inspect it before editing, so I will add the command to the new client.

Reply:
See https://github.com/vitessio/vitess/pull/15389, which I will rebase this on after merging.